### PR TITLE
Callback for PIP mode

### LINKED ISSUE #76: Is there any callback function when video goes to PIP?

### PR DESCRIPTION
**Pull request by jimmyjose-dev**
Listener added to capture Jitsi's entry and exit from PIP mode
Addresses this query -> #76 

Sample usage in dart code..

Add Listener
```

JitsiMeet.addListener(JitsiMeetingListener(
        onConferenceWillJoin: _onConferenceWillJoin,
        onConferenceJoined: _onConferenceJoined,
        onConferenceTerminated: _onConferenceTerminated,
        onPictureInPictureWillEnter: _onPictureInPictureWillEnter,
        onPictureInPictureTerminated: _onPictureInPictureTerminated,
        onError: _onError));
```

Then
```

_onPictureInPictureWillEnter({message}) {
    print("_onPictureInPictureWillEnter");
  }

  _onPictureInPictureTerminated({message}) {
    print("_onPictureInPictureTerminated");
  }
```

### CODE REVIEW

File: android/src/main/kotlin/com/gunschu/jitsi_meet/JitsiMeetPluginActivity.kt [hunk 1]
```diff
@@ -32,10 +33,24 @@ class JitsiMeetPluginActivity : JitsiMeetActivity() {
     override fun onPictureInPictureModeChanged(isInPictureInPictureMode: Boolean, newConfig: Configuration?) {
         super.onPictureInPictureModeChanged(isInPictureInPictureMode, newConfig)
 
+        if (isInPictureInPictureMode){
+
+            JitsiMeetEventStreamHandler.instance.onPictureInPictureWillEnter()
+
+        }else{
+
+
```
Comment (tuantvu) [inline android/src/main/kotlin/com/gunschu/jitsi_meet/JitsiMeetPluginActivity.kt:42]: Let's remove all these extra blank lines.

File: ios/Classes/JitsiViewController.swift [hunk 2]
```diff
@@ -1,7 +1,15 @@
 import UIKit
 import JitsiMeet
 
-class JitsiViewController: UIViewController {
+class JitsiViewController: UIViewController,PiPViewCoordinatorDelegate {
+    
+    func exitPictureInPicture() {
```
Comment (tuantvu) [inline ios/Classes/JitsiViewController.swift:6]: Could  this function be moved to the same extension class as the enterPictureInPicture function?

File: ios/Classes/JitsiViewController.swift [hunk 3]
```diff
@@ -49,7 +57,11 @@ class JitsiViewController: UIViewController {
     
     override func viewWillTransition(to size: CGSize,
                                      with coordinator: UIViewControllerTransitionCoordinator) {
+
```
Comment (tuantvu) [inline ios/Classes/JitsiViewController.swift:60]: Please remove all extra blank lines

File: ios/Classes/JitsiViewController.swift (hunk 2 above)
Comment (tuantvu) [inline ios/Classes/JitsiViewController.swift:6]: Is this function overriding some function in PiPViewCoordinatorDelegate?

File: lib/jitsi_meeting_listener.dart [hunk 4]
```diff
@@ -3,11 +3,15 @@ class JitsiMeetingListener {
...
   final Function({Map<dynamic, dynamic> message}) onConferenceTerminated;
+  final Function({Map<dynamic, dynamic> message}) onPictureInPictureWillEnter;
+  final Function({Map<dynamic, dynamic> message}) onPictureInPictureTerminated;
   final Function(dynamic error) onError;
 
   JitsiMeetingListener(
       {this.onConferenceWillJoin,
       this.onConferenceJoined,
       this.onConferenceTerminated,
+        this.onPictureInPictureWillEnter,
```
Comment (tuantvu) [inline lib/jitsi_meeting_listener.dart:14]: remove extra indent.

**Review by tuantvu — CHANGES_REQUESTED** [review]

**Comment by jimmyjose-dev** [comment]
I will refactor the code and do a PR again.. not sure whats messing up my formatting when I submit.

**Comment by jimmyjose-dev** [comment]
Code updated and merged through another pull request.

**Pull request closed without merging**